server/src/test/java/org/opensearch/cluster/coordination/JoinTaskExecutorTests.java

```diff
@@ -1378,6 +1379,72 @@ public void testJoinRemoteStoreClusterWithRemotePublicationNodeInMixedMode() {
         JoinTaskExecutor.ensureNodesCompatibility(joiningNode, currentState.getNodes(), currentState.metadata());
     }
 
+    public void testUpdatesClusterStateWithRepositoryMetadataNotInSync() throws Exception {
```

Comment:
Based on the stacktrace shared in the issue, the remote nodes are part of cluster-state but the repos are not initialized in RepoService. Could we have it covered in the test ?
```
[2024-12-03T17:32:27,926][INFO ][o.o.c.c.JoinHelper       ] [node_t0] failed to join {node_t0}{HI4LDSiRQserbA7_FDq4Dw}{fWSEQusjSgiDULZF2oBNEw}{127.0.0.1}{127.0.0.1:50328}{m}{shard_indexing_pressure_enabled=true} with JoinRequest{sourceNode={node_t0}{HI4LDSiRQserbA7_FDq4Dw}{fWSEQusjSgiDULZF2oBNEw}{127.0.0.1}{127.0.0.1:50328}{m}{shard_indexing_pressure_enabled=true}, minimumTerm=1, optionalJoin=Optional[Join{term=2, lastAcceptedTerm=1, lastAcceptedVersion=8, sourceNode={node_t0}{HI4LDSiRQserbA7_FDq4Dw}{fWSEQusjSgiDULZF2oBNEw}{127.0.0.1}{127.0.0.1:50328}{m}{shard_indexing_pressure_enabled=true}, targetNode={node_t0}{HI4LDSiRQserbA7_FDq4Dw}{fWSEQusjSgiDULZF2oBNEw}{127.0.0.1}{127.0.0.1:50328}{m}{shard_indexing_pressure_enabled=true}}]}
org.opensearch.transport.RemoteTransportException: [node_t0][127.0.0.1:50328][internal:cluster/coordination/join]
Caused by: java.lang.NullPointerException: Cannot invoke "org.opensearch.repositories.Repository.getRestrictedSystemRepositorySettings()" because "repository" is null
	at org.opensearch.repositories.RepositoriesService.ensureValidSystemRepositoryUpdate(RepositoriesService.java:907) ~[main/:?]
	at org.opensearch.node.remotestore.RemoteStoreNodeService.updateRepositoriesMetadata(RemoteStoreNodeService.java:194) ~[main/:?]
	at org.opensearch.cluster.coordination.JoinTaskExecutor.execute(JoinTaskExecutor.java:210) ~[main/:?]
```

Reply:
Updated the test to assert for this failure condition where in the persisted metadata has the repositories but the repository service does not.